src/compiler.ts

```diff
@@ -812,11 +814,18 @@ export class Compiler extends DiagnosticEmitter {
       case ElementKind.CLASS: {
         let classInstance = <Class>element;
         // make the class name itself represent its runtime id
-        if (!classInstance.type.isUnmanaged) {
+        if (!classInstance.type.isUnmanaged && !this.globalClasses.has(classInstance)) {
```

Comment:
There can be multiple exports with different names of the same class, sharing a single global. As such, this seems not quite correct, in that it also skips `addGlobalExport` but should only skip the `addGlobal` if the class has already been processed.